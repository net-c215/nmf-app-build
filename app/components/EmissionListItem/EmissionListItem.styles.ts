import { StyleSheet } from "react-native";
import { PADDING_HORIZONTAL } from "../../constants/Layout";

export default StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginVertical: 4,
    marginHorizontal: PADDING_HORIZONTAL
  },
  textContainer: {
    flex: 1
  },
  icon: {
    paddingHorizontal: PADDING_HORIZONTAL,
    marginTop: 2
  }
});
