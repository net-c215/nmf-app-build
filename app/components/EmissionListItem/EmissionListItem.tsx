import React from "react";
import { View } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import colors from "../../style/colors";
import styles from "./EmissionListItem.styles";
import Text from "../Text";

interface Props {
  plane?: boolean;
  restaurant?: boolean;
  build?: boolean;
  title: string;
  subTitle: string;
}

const EmissionListItem = ({
  plane = false,
  restaurant = false,
  build = false,
  title = "",
  subTitle = ""
}: Props) => {
  let iconName = "";
  let iconSize = 28;

  if (build) {
    iconName = "md-build";
    iconSize = 26;
  }

  if (plane) {
    iconName = "md-airplane";
    iconSize = 30;
  }

  if (restaurant) {
    iconName = "md-restaurant";
  }

  return (
    <View style={styles.container}>
      <Ionicons
        name={iconName}
        size={iconSize}
        style={styles.icon}
        color={colors.darkLink}
      />
      <View style={styles.textContainer}>
        <Text.Primary numberOfLines={1}>{title}</Text.Primary>
        <Text.Tertiary numberOfLines={1} lightGray>
          {subTitle}
        </Text.Tertiary>
      </View>
      <Ionicons
        name={"ios-arrow-forward"}
        size={22}
        style={styles.icon}
        color={colors.darkLink}
      />
    </View>
  );
};

export default EmissionListItem;
