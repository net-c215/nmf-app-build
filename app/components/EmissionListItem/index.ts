import { EmissionListItem } from "./EmissionListItem";
import { EmissionListItemProps } from "./EmissionListItem";

export { EmissionListItem, EmissionListItemProps };
