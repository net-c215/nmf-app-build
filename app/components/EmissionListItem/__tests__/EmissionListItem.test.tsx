import React from "react";
import renderer from "react-test-renderer";
import EmissionListItem from "../EmissionListItem";

jest.unmock("../EmissionListItem.tsx");

const TITLE = "170 g of red meat";
const SUBTITLE = "2.1 kg CO2";

it("EmissionListItem renders correctly by default", () => {
  const tree = renderer
    .create(<EmissionListItem title={TITLE} subTitle={SUBTITLE} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("EmissionListItem renders correctly with restaurant icon", () => {
  const tree = renderer
    .create(<EmissionListItem title={TITLE} subTitle={SUBTITLE} restaurant />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("EmissionListItem renders correctly with plane icon", () => {
  const tree = renderer
    .create(<EmissionListItem title={TITLE} subTitle={SUBTITLE} plane />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("EmissionListItem renders correctly with build icon", () => {
  const tree = renderer
    .create(<EmissionListItem title={TITLE} subTitle={SUBTITLE} build />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
