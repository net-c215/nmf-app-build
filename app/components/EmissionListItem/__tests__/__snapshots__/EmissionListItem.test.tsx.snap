// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmissionListItem renders correctly by default 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 14,
      "marginVertical": 4,
    }
  }
>
  <Ionicons
    color="#111714"
    name=""
    size={28}
    style={
      Object {
        "marginTop": 2,
        "paddingHorizontal": 14,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text.Primary
      numberOfLines={1}
    >
      170 g of red meat
    </Text.Primary>
    <Text.Tertiary
      lightGray={true}
      numberOfLines={1}
    >
      2.1 kg CO2
    </Text.Tertiary>
  </View>
  <Ionicons
    color="#111714"
    name="ios-arrow-forward"
    size={22}
    style={
      Object {
        "marginTop": 2,
        "paddingHorizontal": 14,
      }
    }
  />
</View>
`;

exports[`EmissionListItem renders correctly with build icon 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 14,
      "marginVertical": 4,
    }
  }
>
  <Ionicons
    color="#111714"
    name="md-build"
    size={26}
    style={
      Object {
        "marginTop": 2,
        "paddingHorizontal": 14,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text.Primary
      numberOfLines={1}
    >
      170 g of red meat
    </Text.Primary>
    <Text.Tertiary
      lightGray={true}
      numberOfLines={1}
    >
      2.1 kg CO2
    </Text.Tertiary>
  </View>
  <Ionicons
    color="#111714"
    name="ios-arrow-forward"
    size={22}
    style={
      Object {
        "marginTop": 2,
        "paddingHorizontal": 14,
      }
    }
  />
</View>
`;

exports[`EmissionListItem renders correctly with plane icon 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 14,
      "marginVertical": 4,
    }
  }
>
  <Ionicons
    color="#111714"
    name="md-airplane"
    size={30}
    style={
      Object {
        "marginTop": 2,
        "paddingHorizontal": 14,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text.Primary
      numberOfLines={1}
    >
      170 g of red meat
    </Text.Primary>
    <Text.Tertiary
      lightGray={true}
      numberOfLines={1}
    >
      2.1 kg CO2
    </Text.Tertiary>
  </View>
  <Ionicons
    color="#111714"
    name="ios-arrow-forward"
    size={22}
    style={
      Object {
        "marginTop": 2,
        "paddingHorizontal": 14,
      }
    }
  />
</View>
`;

exports[`EmissionListItem renders correctly with restaurant icon 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 14,
      "marginVertical": 4,
    }
  }
>
  <Ionicons
    color="#111714"
    name="md-restaurant"
    size={28}
    style={
      Object {
        "marginTop": 2,
        "paddingHorizontal": 14,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text.Primary
      numberOfLines={1}
    >
      170 g of red meat
    </Text.Primary>
    <Text.Tertiary
      lightGray={true}
      numberOfLines={1}
    >
      2.1 kg CO2
    </Text.Tertiary>
  </View>
  <Ionicons
    color="#111714"
    name="ios-arrow-forward"
    size={22}
    style={
      Object {
        "marginTop": 2,
        "paddingHorizontal": 14,
      }
    }
  />
</View>
`;
